src/main/java/org/altbeacon/beacon/BeaconIntentProcessor.java

```diff
@@ -44,6 +47,19 @@ public BeaconIntentProcessor() {
 
 	@Override
 	protected void onHandleIntent(Intent intent) {
+        //Make sure we always deliver results on the main thread
```

Comment:
`onHandleIntent()` always gets executed on a worker thread, as this is the expected behaviour of an `IntentService` (see [documentation](https://developer.android.com/reference/android/app/IntentService.html#onHandleIntent(android.content.Intent))).

It's probably better not to mess with Android callbacks by calling them directly.
It might be safer to instantiate the Handler in the onCreate() and add a new method that posts to the main thread the calls to the listeners.